tests/test_self_speculation_generator.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+#
+
+
+import pytest
+import torch
+
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from tests.tests_constants import local_model_path
+
+from self_speculation.generator_base import GenerationConfig
+from self_speculation.self_speculation_generator import SelfSpeculativeGenerationStrategy
+import logging
+
+@pytest.fixture
+def model_and_config():
+    """Fixture to create a model and a generation configuration."""
+    # Initialize a lightweight version of LlamaForCausalLM if possible
+    tokenizer = AutoTokenizer.from_pretrained(local_model_path)
+    model = AutoModelForCausalLM.from_pretrained(
+        local_model_path,
+        use_safetensors=True,
+        device_map="auto",
+        torch_dtype=torch.float,
+    )
+
+    model.eval()
+    config = GenerationConfig(max_steps=4)
```

Comment:
I did this change locally and it seems to fix the 2 errors.
I think the reason is taht by default `num_speculations` is -1 and hence there are no draft generations and that caused the numerical errors.
```suggestion
    config = GenerationConfig(max_steps=4, exit_layer=4, num_speculations=4)
```